crypto/ec/asm/ecp_nistz256-x86_64.pl

```diff
@@ -3620,35 +3620,16 @@ ()
 	or	$acc0, $acc4
 	or	$acc1, $acc4			# !is_equal(U1, U2)
 
-	movq	%xmm2, $acc0
-	movq	%xmm3, $acc1
+	movq	%xmm2, $acc0			# in1infty | in2infty
+	movq	%xmm3, $acc1			# !is_equal(S1, S2)
 
 	or	$acc0, $acc4
-	.byte	0x3e				# predict taken
-	jnz	.Ladd_proceed$x			# !is_equal(U1, U2) || in1infty || in2infty
-
-	# We now know A = B or A = -B and neither is infinity. Compare the
-	# y-coordinates via S1 and S2.
-	test	$acc1, $acc1
-	jz	.Ladd_double$x			# is_equal(S1, S2)
+	or	$acc1, $acc4
 
-	# A = -B, so the result is infinity.
-	#
-	# TODO: see https://github.com/google/boringssl/blob/12d9ed670da3edd64ce8175cfe0e091982989c18/crypto/fipsmodule/ec/asm/p256-x86_64-asm.pl#L3128-L3132
-	# Does .Ladd_proceed handle this case? It seems to, in
-	# which case we should eliminate this special-case and simplify the
-	# timing analysis.
-	movq	%xmm0, $r_ptr			# restore $r_ptr
-	pxor	%xmm0, %xmm0
-	movdqu	%xmm0, 0x00($r_ptr)
-	movdqu	%xmm0, 0x10($r_ptr)
-	movdqu	%xmm0, 0x20($r_ptr)
-	movdqu	%xmm0, 0x30($r_ptr)
-	movdqu	%xmm0, 0x40($r_ptr)
-	movdqu	%xmm0, 0x50($r_ptr)
-	jmp	.Ladd_done$x
```

Comment:
I see this TODO has not been resolved in boring yet: https://github.com/google/boringssl/blob/master/crypto/fipsmodule/ec/asm/p256-x86_64-asm.pl#L3130-L3132

Is it really safe to remove it?

Reply:
I think so, it is much more easy to understand in the reference implementation.
The case where we add(p, -p), p not infinity we know the following
U1 == U1, S1 != S2 Z1 != 0 Z2 != 0
therefore H == 0
and since Z3 = H * Z1 * Z2 = 0 the result is infinity.
The other coordinates are X3 = R^2, Y3 = -R^3 but they don't matter.
I will try to add a few words about that observation in the comment.

Reply:
Perfect!